backbone.js

```diff
@@ -1140,6 +1140,7 @@
         removed.push(model);
         this._removeReference(model, options);
       }
+      if (models.length > 0 && !options.silent) delete options.index;
```

Comment:
Why the `options.silent` check? If it's just to check whether `add` will fire, shouldn't we always remove it?

Also, you don't have to delete this prop (`delete` is slow). Just set it to `undefined`.